src/lib.rs

```diff
@@ -1,40 +1,100 @@
 // rust-hwid
 // (c) 2020 tilda, under MIT license
-extern crate winreg;
-use winreg::enums::HKEY_LOCAL_MACHINE;
+
+//! Get a "Hardware ID" for the host machine. This is a UUID
+//! which is intended to uniquely represent this entity.
+
+use thiserror::Error;
+
+/// Possible failure cases for [get_id()].
+#[derive(Debug, Error)]
+pub enum HwIdError {
+    /// Could not detect a hardware id. This might be caused
+    /// by a misconfigured system or by this feature not
+    /// being supported by the system or platform.
+    #[error("no HWID was found on system")]
+    NotFound,
+    /// Found a putative HWID, but something was wrong with
+    /// it.  The `String` argument contains a path or other
+    /// identifier at which the HWID was found. This will
+    /// usually indicate something is really wrong with the
+    /// system.
+    #[error("{0:?}: contains malformed HWID")]
+    Malformed(String),
+}
 
 #[cfg(target_os = "windows")]
-pub fn get_id() -> std::string::String {
-    // escaping is fun, right? right???
-    let hive = winreg::RegKey::predef(HKEY_LOCAL_MACHINE).open_subkey("\\\\SOFTWARE\\Microsoft\\Cryptography").expect("Failed to open subkey");
-    let id = hive.get_value("MachineGuid").expect("Failed to get MachineGuid");
-    return id
+mod hwid {
+    use super::*;
+    use winreg::enums::HKEY_LOCAL_MACHINE;
+
+    /// Get the hardware ID of this machine. The HWID is
+    /// obtained from the Windows registry at location
+    /// `\\\\SOFTWARE\\Microsoft\\Cryptography\\MachineGuid`.
+    pub fn get_id() -> Result<std::string::String, HwIdError> {
+        // escaping is fun, right? right???
+        let hive = winreg::RegKey::predef(HKEY_LOCAL_MACHINE)
+            .open_subkey(r"\\SOFTWARE\Microsoft\Cryptography")
```

Comment:
change the argument for `open_subkey` to this: `r"Software\Microsoft\Cryptography"`

apparently despite the key being called "SOFTWARE" in regedit you have to refer to it as "Software". classic windows I guess 🤷.

Reply:
Nice. :shrug:  I guess it's one of those Windows "capitalize eight letters or less" things.

Thanks much, change pushed!